spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -386,6 +386,10 @@ def __init__(self, name=None, plugin=None, parent=None,
 
         self._cached_kernel_properties = None
 
+    def on_close(self):
+        self.mainwindow_close = True
```

Comment:
Is this new attribute really necessary? It seems not.

Reply:
It's used in the `close_client` method: https://github.com/ccordoba12/spyder/blob/73e8beb6ba9773507df28c5dabbbb38df2ac28d1/spyder/plugins/ipythonconsole/widgets/main_widget.py#L1855-L1858